prolog/comment.pl

```diff
@@ -134,23 +140,79 @@
     test(comment_token_rec('<','>',TT),S,[]),
     atom_codes(T,TT).
 
-tester([]).
-tester([X|L]) :-
-    write_term(test(X),[]),
+start(AA) :-
+    (
+        catch(b_getval(a,[N,A]),_,N=0) ->
+          true;
+        N=0
+    ),
+    NN is N + 1,
     (
-        call(X) -> write(' ... OK') ; write(' ... FAIL')
+        N == 0 ->
+          AA = _;
+        AA = A
     ),
-    nl,
-    tester(L).
+    b_setval(a,[NN,AA]).
 
+end(A) :-
+    b_getval(a,[N,A]),
+    NN is N - 1,
+    b_setval(a,[NN,A]).
+    
+a(A) --> 
+  {atom_codes(A,AA)},
```

Comment:
Looks like the indentation is slightly off on these first lines.
```suggestion
   {atom_codes(A,AA)},
```